## Logistic Regression Exercise: Cardiovascular Disease

In this exercise you will use data from [kaggle](https://www.kaggle.com/andrewmvd/heart-failure-clinical-data) to build a model to predict the likelihood of death due to heart failure.

In [16]:
import pandas as pd

In [17]:
heart_failure = pd.read_csv('../datasets/heart_failure/heart_failure_clinical_records_dataset.csv')

In [18]:
heart_failure.head()

age  anaemia  creatinine_phosphokinase  diabetes  ejection_fraction  \
0  75.0        0                       582         0                 20   
1  55.0        0                      7861         0                 38   
2  65.0        0                       146         0                 20   
3  50.0        1                       111         0                 20   
4  65.0        1                       160         1                 20   

   high_blood_pressure  platelets  serum_creatinine  serum_sodium  sex  \
0                    1  265000.00               1.9           130    1   
1                    0  263358.03               1.1           136    1   
2                    0  162000.00               1.3           129    1   
3                    0  210000.00               1.9           137    1   
4                    0  327000.00               2.7           116    0   

   smoking  time  DEATH_EVENT  
0        0     4            1  
1        0     6            1  
2        1     7            1  
3        0     7            1  
4        0     8            1

The available features are as follows:

| Feature         |       	Explanation            |                        	Measurement   | 	Range     |
|-----|------|------|-----|         
|Age  |                  	Age of the patient     |                       	Years      |    	[40,…, 95]   |   
| Anaemia         |       	Decrease of red blood cells or hemoglobin   |   	Boolean      |  	0, 1  |            
| High blood pressure    |	If a patient has hypertension    |  	Boolean     |   	0, 1    |          
|Creatinine phosphokinase (CPK)	| Level of the CPK enzyme in the blood      |    	mcg/L      |    	[23,…, 7861]    
| Diabetes           |     	If the patient has diabetes          |          	Boolean    |    	0, 1     |         
| Ejection fraction     | 	Percentage of blood leaving          |          	Percentage     | 	[14,…, 80] |     
| Sex        |            	Woman or man              |                   	Binary    |      	0, 1    |          
| Platelets     |         	Platelets in the blood               |         	kiloplatelets/mL  |	[25.01,…, 850.00]  |
| Serum creatinine   |     	Level of creatinine in the blood     |         	mg/dL     |     	[0.50,…, 9.40] | 
| Serum sodium     |       	Level of sodium in the blood      |            	mEq/L     |     	[114,…, 148] |   
| Smoking            |    	If the patient smokes   |                       	Boolean        |	0, 1   |           
| Time          |          	Follow-up period       | 	Days         |   	[4,…,285]      |  
| (target) death event  |  	If the patient died during the follow-up period	 |  Boolean   |     	0, 1   |       

**Step 1:** Using all features **besides time**, fit a logistic regression model to predict the last columns, DEATH_EVENT.

In [1]:
# Your code here

What accuracy does your model have? 

Which metric (**precision** or **recall**) would you use to answer this question: "Out of those patients that died, the model correctly predicted what percentage?"

Which metric (**precision** or **recall**) would you use to answer this question: "Out of those patients that the model predicted would die, what percentage died?"

Take a look at the coefficients of the model. For which featues did an increase tend to be associated with an increased likelihood of death (holding all others constant)?

In [2]:
# Your code here

Now, try out regularization to find the best model possible. **Hint:** regularization is controlled by the penalty and C arguments in the LogisticRegression class.

In [3]:
# Your code here